src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -21,6 +22,168 @@
 
 namespace Stratis.Bitcoin.Consensus
 {
+    internal class BlockIndexer
+    {
```

Comment:
I think both dictionaries are not needed, for `blocksById ` we can reuse the dictionary in CHT
And for `blocksByHeight ` we can just leverage the count of that same dictionary (genesis is at position zero anyway so there is not even an offset by 1)

Reply:
yes basically `private readonly Dictionary<int, ChainedHeader> blocksByHeight` could be a
`private readonly List<ChainedHeader> blocksByHeight` where the index is the height
we have just to be sure to populate it in order